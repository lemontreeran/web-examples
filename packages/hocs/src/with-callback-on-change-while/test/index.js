import React from 'react'
import { mount } from 'enzyme'

import withCallbackOnChangeWhile from '../src/'

const Target = () => null

describe('withCallbackOnChangeWhile', () => {
  it('should pass props through', () => {
    const EnhancedTarget = withCallbackOnChangeWhile()(Target)
    const wrapper = mount(
      <EnhancedTarget a={1} b={2} c={3} />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('should invoke a callback on prop change and match condition', () => {
    const mockShouldCall = jest.fn(() => false).mockImplementationOnce(() => true)
    const mockCallback = jest.fn()
    const EnhancedTarget = withCallbackOnChangeWhile(
      'a',
      mockShouldCall,
      mockCallback
    )(Target)
    const wrapper = mount(
      <EnhancedTarget a={1} b={2} c={3} />
    )

    expect(mockShouldCall.mock.calls).toMatchSnapshot()
    expect(mockCallback.mock.calls).toMatchSnapshot()
    wrapper.setProps({ a: 11 })
    expect(mockShouldCall.mock.calls).toMatchSnapshot()
    expect(mockCallback.mock.calls).toMatchSnapshot()
    wrapper.setProps({ a: 111 })
    expect(mockShouldCall.mock.calls).toMatchSnapshot()
    expect(mockCallback.mock.calls).toMatchSnapshot()
  })

  it('should no-op if prop is the same', () => {
    const mockShouldCall = jest.fn()
    const mockCallback = jest.fn()
    const EnhancedTarget = withCallbackOnChangeWhile(
      'a',
      mockShouldCall,
      mockCallback
    )(Target)
    const wrapper = mount(
      <EnhancedTarget a={1} b={2} c={3} />
    )

    expect(mockShouldCall.mock.calls).toMatchSnapshot()
    expect(mockCallback.mock.calls).toMatchSnapshot()
    wrapper.setProps({ a: 1 })
    expect(mockShouldCall.mock.calls).toMatchSnapshot()
    expect(mockCallback.mock.calls).toMatchSnapshot()
  })

  describe('display name', () => {
    const origNodeEnv = process.env.NODE_ENV

    afterAll(() => {
      process.env.NODE_ENV = origNodeEnv
    })

    it('should wrap display name in non-production env', () => {
      process.env.NODE_ENV = 'test'

      const EnhancedTarget = withCallbackOnChangeWhile()(Target)
      const wrapper = mount(
        <EnhancedTarget />
      )

      expect(wrapper).toMatchSnapshot()
    })

    it('should not wrap display name in production env', () => {
      process.env.NODE_ENV = 'production'

      const EnhancedTarget = withCallbackOnChangeWhile()(Target)
      const wrapper = mount(
        <EnhancedTarget />
      )

      expect(wrapper).toMatchSnapshot()
    })
  })
})
