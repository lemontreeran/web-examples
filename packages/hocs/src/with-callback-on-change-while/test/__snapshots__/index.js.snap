// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withCallbackOnChangeWhile display name should not wrap display name in production env 1`] = `
<WithCallbackOnChangeWhile>
  <Target />
</WithCallbackOnChangeWhile>
`;

exports[`withCallbackOnChangeWhile display name should wrap display name in non-production env 1`] = `
<withCallbackOnChangeWhile(Target)>
  <Target />
</withCallbackOnChangeWhile(Target)>
`;

exports[`withCallbackOnChangeWhile should invoke a callback on prop change and match condition 1`] = `Array []`;

exports[`withCallbackOnChangeWhile should invoke a callback on prop change and match condition 2`] = `Array []`;

exports[`withCallbackOnChangeWhile should invoke a callback on prop change and match condition 3`] = `
Array [
  Array [
    Object {
      "a": 11,
      "b": 2,
      "c": 3,
    },
  ],
]
`;

exports[`withCallbackOnChangeWhile should invoke a callback on prop change and match condition 4`] = `
Array [
  Array [
    Object {
      "a": 11,
      "b": 2,
      "c": 3,
    },
  ],
]
`;

exports[`withCallbackOnChangeWhile should invoke a callback on prop change and match condition 5`] = `
Array [
  Array [
    Object {
      "a": 11,
      "b": 2,
      "c": 3,
    },
  ],
  Array [
    Object {
      "a": 111,
      "b": 2,
      "c": 3,
    },
  ],
]
`;

exports[`withCallbackOnChangeWhile should invoke a callback on prop change and match condition 6`] = `
Array [
  Array [
    Object {
      "a": 11,
      "b": 2,
      "c": 3,
    },
  ],
]
`;

exports[`withCallbackOnChangeWhile should no-op if prop is the same 1`] = `Array []`;

exports[`withCallbackOnChangeWhile should no-op if prop is the same 2`] = `Array []`;

exports[`withCallbackOnChangeWhile should no-op if prop is the same 3`] = `Array []`;

exports[`withCallbackOnChangeWhile should no-op if prop is the same 4`] = `Array []`;

exports[`withCallbackOnChangeWhile should pass props through 1`] = `
<withCallbackOnChangeWhile(Target)
  a={1}
  b={2}
  c={3}
>
  <Target
    a={1}
    b={2}
    c={3}
  />
</withCallbackOnChangeWhile(Target)>
`;
